python/idsse_common/idsse/common/http_utils.py

```diff
@@ -0,0 +1,90 @@
+"""Helper function for listing directories and retrieving s3 objects"""
+# -------------------------------------------------------------------------------
+# Created on Tue Dec 3 2024
+#
+# Copyright (c) 2023 Colorado State University. All rights reserved.             (1)
+# Copyright (c) 2023 Regents of the University of Colorado. All rights reserved. (2)
+#
+# Contributors:
+#     Paul Hamer (1)
+#
+# -------------------------------------------------------------------------------
+import logging
+import os
+import shutil
+from collections.abc import Sequence
+
+import requests
+
+from .protocol_utils import ProtocolUtils
+
+logger = logging.getLogger(__name__)
+
+# pylint: disable=duplicate-code, broad-exception-caught
+
+class HttpUtils(ProtocolUtils):
+    """http Utility Class - Used by DAS for file downloads"""
+
+    def ls(self, path: str, prepend_path: bool = True) -> Sequence[str]:
+        """Execute a 'ls' on the AWS s3 bucket specified by path
+
+        Args:
+            path (str): s3 bucket
+            prepend_path (bool): Add to the filename
+
+        Returns:
+            Sequence[str]: The results sent to stdout from executing a 'ls' on passed path
+        """
+        return self.http_ls(path, prepend_path)
+
+
+    @staticmethod
+    def http_ls(url: str, prepend_path: bool = True) -> Sequence[str]:
```

Comment:
Similar to aws_utils, just have 'ls' and 'cp' implement abstract methods directly.